extension.js

```diff
@@ -112,9 +113,58 @@ function description(app) {
     return appName + ' → ' + app.get_title();
 }
 
-function updateHighlight(boxes) {
-  boxes.forEach(box => box.whole.remove_style_class_name('switcher-highlight'));
-  boxes.length > cursor && boxes[cursor].whole.add_style_class_name('switcher-highlight');
+function escapeChars(text) {
+    return text.replace(/[-[\]{}()*+?.,\\^$|#]/g, "\\$&");
+};
+
+function highlightText(text, query) {
+  // Don't apply highlighting if there's no input
+  if (query == "")
+    return text;
+
+  // Escape special characters in query
+  query = escapeChars(query);
+
+  // Identify substring parts to be highlighted
+  let queryExpression = "(";
+  let queries = query.split(' ');
+  let queriesLength = queries.length;
+  for (let i = 0; i < queriesLength - 1; i++) {
+    if (queries[i] != "") {
+      queryExpression += queries[i] + "|";
+    }
+  }
+  queryExpression += queries[queriesLength - 1] + ")";
+  let queryRegExp = new RegExp(queryExpression, "i");
+  let tokenRegExp = new RegExp("^" + queryExpression + "$", "i");
+
+  // Build resulting string from highlighted and non-highlighted strings
+  let result = "";
+  let tokens = text.split(queryRegExp);
+  let tokensLength = tokens.length;
+  for (let i = 0; i < tokensLength; i++) {
+    if (tokens[i].match(tokenRegExp)) {
+      result += '<span background=\"#4a90d9\" foreground=\"#ffffff\">' +
```

Comment:
Continuing the conversation from the other commit. How often do you really have titles with underscores? To my eye the underline under underscore, while it looks a bit silly, does work. Another option might be to change the default font to be thinner and then bold the matching parts.

![switcher-underscore](https://cloud.githubusercontent.com/assets/1044056/13020439/0f202668-d1de-11e5-8453-621cda563be0.png)